### DISCUSSION TITLE
[unocss support] How to configure dark mode with unocss?

### DISCUSSION THREAD

**Discussion by spokospace** [withastro/starlight#1855] [Q&A] [answered]
Hello, is it possible to add the 'dark' class to the body? 
It is required to display the styles correctly on [unocss](https://unocss.dev/integrations/astro).

**Comment by delucis** [comment, marked as answer]
Starlight uses a `data-theme` attribute on the `<html>` element to indicate the current theme, i.e.

```html
<html data-theme="dark">
<!-- or -->
<html data-theme="light">
```

I’ve never used UnoCSS, but based on their [`dark` configuration reference](https://unocss.dev/presets/mini#dark), I think you should be able to configure it to use Starlight’s existing selectors:

```js
presetMini({
  dark: {
    dark: '[data-theme="dark"]',
    light: '[data-theme="light"]',
  },
})
```

**Comment by spokospace** [comment]
I noticed issues with code like `class="dark:invert"`, unocss is based on Tailwind, but with more useful features like grouping etc. That's pity there is no option to configure starlight or modify html.

I wanted to try with creating documentation in astro, but for me as a front-end developer VitePress is more friendly and has the necessary functionality right away:
![image](https://github.com/withastro/starlight/assets/16397823/fafdb545-2b19-466e-8642-f07c000b7ab9)


edit:

okay, I managed to fix the selector, I guess the order of the presets is important, my current code:

```
// uno.config.ts
import {
  defineConfig,
  transformerVariantGroup,
  presetAttributify,
  presetIcons,
  presetTypography,
  presetUno,
  presetWebFonts,
  transformerDirectives,
} from 'unocss'

...
export default defineConfig({

transformers: [
  transformerDirectives(),
  transformerVariantGroup(),
],

theme: {
	...
}

presets: [
 presetUno(
      {
        dark: { 
          dark: '[data-theme="dark"]',
        }
      }
    ),
    presetAttributify(),
    presetIcons({
      scale: 1.2,
      warn: true,
      prefix: 'i-',
      extraProperties: {
        'display': 'inline-block',
        'vertical-align': 'middle',
      },
    }),
    presetTypography(),
	...
]

})
```

so the fix is here: ```
    presetUno(
      {
        dark: { 
          dark: '[data-theme="dark"]'
        }
      }
    )
    ```

**Reply by delucis** [reply]
Thanks for sharing!
